src/middlewared/middlewared/plugins/crypto.py

```diff
@@ -831,20 +1113,58 @@ def __create_internal(self, data):
         return await self._get_instance(id)
 
     @accepts(
-        Int('id')
+        Int('id'),
+        Bool('force', default=False)
     )
-    async def do_delete(self, id):
-        response = await self.middleware.call(
+    @job(lock='cert_delete')
+    def do_delete(self, job, id, force=False):
+
+        cert = self.middleware.call_sync(
+            'certificate.query',
+            [
+                ['id', '=', id],
+                ['acme', '!=', None]
+            ]
+        )
+
+        if cert:
+            cert = cert[0]
+
+            verrors = ValidationErrors()
+
+            client, key = self.get_acme_client_and_key(cert['acme']['directory'], True)
+
+            job.set_progress(60)
+
+            try:
+                client.revoke(
+                    jose.ComparableX509(crypto.load_certificate(crypto.FILETYPE_PEM, cert['certificate'])),
+                    0
+                )
+            except errors.ClientError as e:
+                verrors.add(
+                    'acme_revoke.id',
```

Comment:
Why acme_revoke.id? Where is acme_revoke defined in schema?

I believe it should be a CallError, not a validation error, nothing failed to validate here.